src/pages/Videos/VideoDetail.js

```diff
@@ -0,0 +1,120 @@
+import { useNavigate, useParams } from "react-router";
+import { Iframe } from "../../components/Iframe";
+import { ToggleIconGroup } from "./VideoListing";
+import { Avatar } from "../../components/BaseCard";
+import { SaveModalButton, PlaylistModal } from "../../components/PlaylistModal";
+import { useEffect, useState } from "react";
+import { usePlaylists } from "../Playlists/playlists-context";
+import {
+  ADD_TO_HISTORY,
+  SET_HISTORY,
+  UPDATE_TIMESTAMP,
+} from "../History/history-reducer";
+import { Notes } from "./Notes";
+import { useAxios } from "../../custom hooks/useAxios";
+import { API_HISTORY, API_VIDEOS } from "../../urls";
+import { LoadingIndicator } from "../../components/LoadingIndicator";
+import { getDefaultPlaylistArray } from "../../utils";
+import { useHistory } from "../History/history-context";
+
+export function VideoDetail() {
+  const { videoId } = useParams();
+  const { likedVideosId, watchLaterId, playlists } = usePlaylists();
+  const [showSaveModal, setShowSaveModal] = useState(false);
+  const [video, setVideo] = useState({});
+  const { isLoading, getData: getVideo } = useAxios(`${API_VIDEOS}/${videoId}`);
+  const { postData: addToHistory, getData: getHistory } = useAxios(API_HISTORY);
+  const likedVideos = getDefaultPlaylistArray(playlists, likedVideosId);
+  const watchLater = getDefaultPlaylistArray(playlists, watchLaterId);
+  const { history, historyDispatch } = useHistory();
+  const navigate = useNavigate();
+  const {
+    _id: mongooseVideoId,
+    id,
+    avatarSrc,
+    uploadedBy,
+    finalNotes,
+    name,
+  } = video;
+
+  useEffect(() => {
+    (async () => {
+      const fetchedVideo = await getVideo();
+      if (fetchedVideo) {
+        setVideo(fetchedVideo);
+      } else {
+        navigate("/page-not-found");
+      }
+    })();
+    return () => {
+      setVideo({});
+    };
+  }, []);
+
+  useEffect(() => {
+    if (mongooseVideoId) {
+      if (history.length === 0) {
+        (async () => {
```

Comment:
I feel it is getting a bit verbose here. Try extracting it to a separate function.

Reply:
Good Idea. Created multiple functions for this